rust/nasl-interpreter/src/built_in_functions/ssh.rs

```diff
@@ -0,0 +1,2107 @@
+                                session_id
+                            ));
+                        }
+                    }
+                    Err(_) => {
+                        return Err(FunctionErrorKind::Diagnostic(
+                            format!(
+                                "Failed setting user authentication for SessionID {}",
+                                session_id
+                            ),
+                            Some(NaslValue::Null),
+                        ));
+                    }
+                };
+            }
+
+            /* Our strategy for kbint is to send the password to the first
+            prompt marked as non-echo.  */
+            if password.is_some() && methods.contains(AuthMethods::INTERACTIVE) {
+                loop {
+                    match session.session.userauth_keyboard_interactive(None, None) {
+                        Ok(AuthStatus::Info) => {
+                            let info = match session.session.userauth_keyboard_interactive_info() {
+                                Ok(i) => i,
+                                Err(_) => {
+                                    return Err(FunctionErrorKind::Diagnostic(
+                                        format!(
+                                            "Failed setting user authentication for SessionID {}",
+                                            session_id
+                                        ),
+                                        Some(NaslValue::Null),
+                                    ));
+                                }
+                            };
+                            if verbose {
+                                ctx.logger().info(&format!("SSH kbdint name={}", info.name));
+                                ctx.logger()
+                                    .info(&format!("SSH kbdint instruction{}", info.instruction));
+                            }
+
+                            let mut answers: Vec<String> = Vec::new();
+                            for p in info.prompts.into_iter() {
+                                if !p.echo {
+                                    answers.push(password.unwrap_or_default().to_string());
+                                } else {
+                                    answers.push(String::new());
+                                };
+                            }
+                            match session
+                                .session
+                                .userauth_keyboard_interactive_set_answers(&answers)
+                            {
+                                Ok(_) => {
+                                    return Ok(NaslValue::Number(0));
+                                }
+                                Err(_) => break,
+                            }
+                        }
+                        Ok(_) => {
+                            if verbose {
+                                ctx.logger()
```

Comment:
because this is use an ENV variable to switch it on, different from the general debug level. It will only be available if it is set via, which also turns the libssh debug log on     
```
let verbose = env::var("OPENVAS_LIBSSH_DEBUG")